lib_eio_luv/eio_luv.ml

```diff
@@ -81,6 +81,17 @@ let rec cstructv_to_luv = function
     let x1, x2 = Cstruct.split x max_luv_buffer_size in
     Cstruct.to_bigarray x1 :: cstructv_to_luv (x2 :: xs)
 
+let float_of_timespec (ts : Luv.File.Stat.timespec) =
+  let s : float = Signed.Long.to_int64  ts.sec |> Int64.to_float in
+  let nsec : float = Signed.Long.to_int64  ts.nsec |> Int64.to_float in
+
+  let n = nsec /. 1_000_000_000.0 in
+  let t = s +. n in
+  if t = (s +. 1.0) then
+    Float.next_after t s
```

Comment:
What's this for? Needs a comment.

Reply:
I basically copied the algorithm from the runtime (they do it in C though), I didn't reason much about it.